### DISCUSSION TITLE
FOLDERS PATH OF INBUND AND OUDBOUND FILES IN OPEN AS2 IN LINUX

### DISCUSSION THREAD

**Discussion by Chetan-Khopade** [OpenAS2/OpenAs2App#334] [Ideas]
Hi Guys,

Now, I've setup new openas2 server version 3.6.0 with webUI and created some Partners configuration and connection for testing but i am not able to see the data location of the Partners, connections, certificates, outbound and inbound files folder path inside of the am Linux.
Could you please help me to where i can found the path of the data folders inside in Linux machine.

**Comment by igwtech** [comment]
Hi,
   Those are configured in the config.xml file as defined in the manual.
   The default location for data files are defined in the
"openas2.properties.sample" file.
-J

On Wed, Aug 16, 2023 at 9:30 AM Chetan-Khopade ***@***.***>
wrote:

> Hi Guys,
>
> Now, I've setup new openas2 server version 3.6.0 with webUI and created
> some Partners configuration and connection for testing but i am not able to
> see the data location of the Partners, connections, certificates, outbound
> and inbound files folder path inside of the am Linux.
> Could you please help me to where i can found the path of the data folders
> inside in Linux machine.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenAS2/OpenAs2App/discussions/334>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA2QND5O2IWXEYBMJ3XXAYTXVTDPLANCNFSM6AAAAAA3SRO4XU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Javier Munoz

CEO

Greicodex Software



+58 2127629120 <+58+2127629120>
***@***.***
www.greicodex.com
Av Francisco Solano, Centro Solano Plaza, Oficina PHA, Caracas, Venezuela

**Comment by uhurusurfa** [comment]
The "storageBaseDir" property defines the base location for inbound and outbound files in a default deployment.

Below is the output running this command **grep storageBaseDir Server/src/config/config.xml** showing all the attributes that define foilder locations:
```
   <properties storageBaseDir="%home%/../data"
               module.MDNFileModule.filename="$properties.storageBaseDir$/$mdn.msg.sender.as2_id$-$mdn.msg.receiver.as2_id$/mdn/$date.yyyy-MM-dd$/$mdn.msg.headers.message-id$"
               module.MessageFileModule.filename="$properties.storageBaseDir$/$msg.sender.as2_id$-$msg.receiver.as2_id$/inbox/$msg.headers.message-id$"
               module.MessageFileModule.header="$properties.storageBaseDir$/$msg.sender.as2_id$-$msg.receiver.as2_id$/msgheaders/$date.yyyy-MM-dd$/$msg.headers.message-id$"
               pollerConfigBase.outboxdir="$properties.storageBaseDir$/outbox/$partnership.receiver.as2_id$"
               pollerConfigBase.errordir="$properties.storageBaseDir$/outbox/error/$date.YYYY$-$date.MM$-$date.dd$/$partnership.receiver.as2_id$"
              pendingMDN="$properties.storageBaseDir$/pendingMDN3"
              pendingMDNinfo="$properties.storageBaseDir$/pendinginfoMDN3"
              outboxdir="$properties.storageBaseDir$/toAny"
              errordir="$properties.storageBaseDir$/toAny/error"
              outboxdir="$properties.storageBaseDir$/toPartnerA/"
              errordir="$properties.storageBaseDir$/toPartnerA/error/$date.YYYY$/$date.MM$"
              sentdir="$properties.storageBaseDir$/$msg.sender.as2_id$-$msg.receiver.as2_id$/sent/$date.YYYY$/$date.MM$"
              tempdir="$properties.storageBaseDir$/temp"/>
              tempdir="$properties.storageBaseDir$/temp"/>
              errordir="$properties.storageBaseDir$/inbox/error"
              errordir="$properties.storageBaseDir$/inbox/error"
              resenddir="$properties.storageBaseDir$/resend"
              errordir="$properties.storageBaseDir$/resend/error"
```

**Comment by Chetan-Khopade** [comment]
Guys,
Thanks for the replay but I would like to tell you that, I've found the one setting for how to create partnerships folder inside of data folder path.
We have to put this line  "pollerConfig enabled="true"    manually in  "partnerships.xml"  file whenever create new partner configuration via WEB GUI but still this is not good way.

So is there anyway via WEB GUI to create the new partnerships folders automatically after creating new partner config?

**Comment by uhurusurfa** [comment]
Currently the WebUI app does not support the **_pollerConfig.enabled_** property sending a value of "**_true_**"necessary to set up a partnership for sending files.
In theory this should be a simple enough enhancement in the WebUI app because it is already supported on the server side but you will have to talk to @GreicodexJM about that enhancement as he dceveloped it.
